.github/workflows/run-tests.yml

```diff
@@ -1,10 +1,26 @@
-# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
 name: Run Tests
 
 on: [push, pull_request]
 
 jobs:
   call_run_tests:
     uses: yext/slapshot-reusable-workflows/.github/workflows/run_tests.yml@v1
+    with:
+      node_matrix: '["16.x"]'
+
+  # It's necessary to use v12 of the react-testing-library since v13 was updated to only support react 18
+  call_run_tests-react-17:
+    uses: yext/slapshot-reusable-workflows/.github/workflows/run_tests.yml@v1
+    with:
+      build_script: |
+        npm i -D react@17 react-dom@17 @testing-library/react@12
+        npm run build
+      node_matrix: '["16.x"]'
+
+  call_run_tests-react-16:
+    uses: yext/slapshot-reusable-workflows/.github/workflows/run_tests.yml@v1
+    with:
+      build_script: |
+        npm i -D react@16.14 react-dom@16.14 @testing-library/react@12
+        npm run build
+      node_matrix: '["16.x"]'
```

Comment:
I thought we still support node versions older than 16?

Reply:
It felt a little egregious to run the unit tests 2 * 3 * 3 = 18 times. The first 2 is for on push/pull_request, the 3 is for 3 node versions, and the last 3 is for 3 react versions. We could take some kind of middle ground where only the react 18 tests are run for 3 node versions, but I think for headless-react it's probably fine to run the tests 18 times since the library is so small and probably will never need to do many other kinds of checks anyways.

tldr the tests run 18 times on prs now!